pkg/provider/conjur/provider.go

```diff
@@ -0,0 +1,159 @@
+type Client interface {
+	RetrieveSecret(secret string) (result []byte, err error)
+}
+
+// NewClient creates a new Conjur client.
+func (p *Provider) NewClient(ctx context.Context, store esv1beta1.GenericStore, kube client.Client, namespace string) (esv1beta1.SecretsClient, error) {
+	cfg, err := getProvider(store)
+	if err != nil {
+		return nil, err
+	}
+	config := conjurapi.Config{
+		Account:      *cfg.ServiceAccount,
+		ApplianceURL: *cfg.ServiceURL,
+		SSLCert:      *cfg.ServiceCertificate,
+	}
+	conjur, err := conjurapi.NewClientFromKey(config,
+		authn.LoginPair{
+			Login:  *cfg.ServiceUser,
+			APIKey: *cfg.ServiceAPIKey,
+		},
+	)
+
+	if err != nil {
+		return nil, fmt.Errorf(errConjurClient, err)
+	}
+	p.ConjurClient = conjur
+	return p, nil
+}
+
+func getProvider(store esv1beta1.GenericStore) (*esv1beta1.ConjurProvider, error) {
+	if store == nil {
+		return nil, errMissingStore
+	}
+	spc := store.GetSpec()
+	if spc == nil || spc.Provider == nil || spc.Provider.Conjur == nil {
+		return nil, errMissingConjurProvider
+	}
+	return spc.Provider.Conjur, nil
+}
+
+// GetAllSecrets returns all secrets from the provider.
+// NOT IMPLEMENTED
+func (p *Provider) GetAllSecrets(ctx context.Context, ref esv1beta1.ExternalSecretFind) (map[string][]byte, error) {
+	// TO be implemented
+	return nil, fmt.Errorf("GetAllSecrets not implemented")
+}
+
+// GetSecret returns a single secret from the provider.
+func (p *Provider) GetSecret(ctx context.Context, ref esv1beta1.ExternalSecretDataRemoteRef) ([]byte, error) {
+	secretValue, err := p.ConjurClient.RetrieveSecret(ref.Key)
+	if err != nil {
+		return nil, err
+	}
+
+	return secretValue, nil
+}
+
+// GetSecretMap returns multiple k/v pairs from the provider.
+func (p *Provider) GetSecretMap(ctx context.Context, ref esv1beta1.ExternalSecretDataRemoteRef) (map[string][]byte, error) {
+	// Gets a secret as normal, expecting secret value to be a json object
+	data, err := p.GetSecret(ctx, ref)
+	if err != nil {
+		return nil, fmt.Errorf("error getting secret %s: %w", ref.Key, err)
+	}
+
+	// Maps the json data to a string:string map
+	kv := make(map[string]string)
+	err = json.Unmarshal(data, &kv)
+	if err != nil {
+		return nil, fmt.Errorf("unable to unmarshal secret %s: %w", ref.Key, err)
+	}
+
+	// Converts values in K:V pairs into bytes, while leaving keys as strings
+	secretData := make(map[string][]byte)
+	for k, v := range kv {
+		secretData[k] = []byte(v)
+	}
+	return secretData, nil
+}
+
+// Close closes the provider.
+func (p *Provider) Close(ctx context.Context) error {
+	return nil
+}
+
+// Validate validates the provider.
+func (p *Provider) Validate() (esv1beta1.ValidationResult, error) {
+	return esv1beta1.ValidationResultReady, nil
+}
+
+// ValidateStore validates the store.
+func (p *Provider) ValidateStore(store esv1beta1.GenericStore) error {
+	storeSpec := store.GetSpec()
+	conjurSpec := storeSpec.Provider.Conjur
+
+	if *conjurSpec.ServiceURL == "" {
```

Comment:
The nil check is missing: `if conjurSpec.ServiceURL != nil && *conjurSpec.ServiceURL == ""`
Otherwise you have a nil pointer dereference panic and the controller dies.

I would suggest switching to a `string` value instead of using `*string` for all the fields.